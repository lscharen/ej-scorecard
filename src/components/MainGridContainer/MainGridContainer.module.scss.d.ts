declare namespace MainGridContainerNamespace {
  export interface IMainGridContainerScss {
    mainGridContainerContainer: string;
  }
}

declare const MainGridContainerScssModule: MainGridContainerNamespace.IMainGridContainerScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: MainGridContainerNamespace.IMainGridContainerScss;
};

export = MainGridContainerScssModule;
