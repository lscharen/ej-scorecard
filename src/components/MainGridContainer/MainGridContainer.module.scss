// @use "uswds-core" as *;

// .mainGridContainerContainer{

// }
