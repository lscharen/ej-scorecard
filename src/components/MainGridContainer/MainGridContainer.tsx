import { ReactNode } from 'react';
import { GridContainer } from '@trussworks/react-uswds';

// import * as styles from './MainGridContainer.module.scss';

export interface IMainGridContainerProps {
  children: ReactNode;
  fullWidth?: boolean;
  className?: string;
  dataCy?: string;
}

const MainGridContainer = ({
  children,
  fullWidth = false,
  className = ``,
  dataCy,
}: IMainGridContainerProps) => {
  return fullWidth ? (
    <div className={className}>{children}</div>
  ) : (
    <GridContainer
      containerSize={`desktop-lg`}
      className={className}
      data-cy={dataCy}
    >
      {children}
    </GridContainer>
  );
};

export default MainGridContainer;
