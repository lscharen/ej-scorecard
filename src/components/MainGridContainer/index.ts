import MainGridContainer from './MainGridContainer';
export default MainGridContainer;
