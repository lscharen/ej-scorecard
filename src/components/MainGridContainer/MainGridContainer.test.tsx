import { render } from '@testing-library/react';
import { IntlProviderTestWrapper } from '@/test/testHelpers';
import MainGridContainer from './MainGridContainer';

describe(`rendering of MainGridContainer Component`, () => {
  const Component = () => <h1>Test</h1>;

  it(`checks if component renders with Full width true`, () => {
    const { asFragment } = render(
      <IntlProviderTestWrapper>
        <MainGridContainer fullWidth={true}>
          <Component />
        </MainGridContainer>
      </IntlProviderTestWrapper>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it(`checks if component renders with Full width false`, () => {
    const { asFragment } = render(
      <IntlProviderTestWrapper>
        <MainGridContainer fullWidth={false}>
          <Component />
        </MainGridContainer>
      </IntlProviderTestWrapper>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
