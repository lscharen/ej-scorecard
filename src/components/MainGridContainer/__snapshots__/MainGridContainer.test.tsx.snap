// Vitest Snapshot v1

exports[`rendering of MainGridContainer Component > checks if component renders with Full width false 1`] = `
<DocumentFragment>
  <div
    class="grid-container-desktop-lg"
    data-testid="gridContainer"
  >
    <h1>
      Test
    </h1>
  </div>
</DocumentFragment>
`;

exports[`rendering of MainGridContainer Component > checks if component renders with Full width true 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <h1>
      Test
    </h1>
  </div>
</DocumentFragment>
`;
